usm_wrappers/salt_wrapper.py

```diff
@@ -63,6 +73,53 @@ def accept_minion(minion_id):
     return (True if out else False)
 
 
+def get_started_minions(minions=[], timeout=60):
+    minion_set = set(minions)
+    up_minions = set(runner.cmd('manage.up', []))
+    if minion_set:
+        started_minions = up_minions.intersection(minion_set)
+    else:
+        started_minions = up_minions
+
+    if not timeout:
+        return started_minions
+
+    time_spent = 0
+    while True:
+        wait = timeout - time_spent
+        if wait <= 0:
+            break
+        start_time = time.time()
+
+        ret = sevent.get_event(wait=wait, full=True, tag='salt/minion')
+
+        end_time = time.time()
+        time_spent += (end_time - start_time)
+
+        if ret is None:
+            continue
+
+        if not fnmatch.fnmatch(ret['tag'], 'salt/minion/*/start'):
+            continue
+
+        minion = ret['data']['id']
+
+        if minion_set:
+            if minion in minion_set:
+                started_minions.add(minion)
+                if started_minions == minion_set:
+                    return started_minions
+        else:
+            started_minions.add(minion)
+
+    # we might have missed out some start events
```

Comment:
Doing this twice seems wrong and a great place for abstraction if you do intend to do it twice.